File: 0128-longest-consecutive-sequence/0128-longest-consecutive-sequence.ts
function longestConsecutive(nums: number[]): number {

    if (nums.length === 0) return 0;

    nums = nums.sort((a, b) => a - b);

    let len: number = 1;

    const arr: number[] = [];

    for (let i = 0; i < nums.length - 1; i++) {

        if (nums[i] + 1 === nums[i + 1]) {

            len++;

        } else if (nums[i] === nums[i + 1]) {

            continue;

        } else {

            arr.push(len);

            len = 1;

        }

    }

    arr.push(len);

    return Math.max(...arr);

};

File: 0128-longest-consecutive-sequence/0128-longest-consecutive-sequence.java

class Solution {

    public int longestConsecutive(int[] nums) {

        if (nums.length == 0 ) return 0;

        Arrays.sort(nums);

        int len = 1;

        HashSet<Integer> arr = new HashSet<>();

        for (int i = 0; i < nums.length - 1; i++) {

            if ( nums[i] + 1 == nums[i + 1] ) {
                len++;
            } else if ( nums[i] == nums[i + 1] ) {
                continue;
            } else {
                arr.add(len);
                len = 1;
            }

        }

        arr.add(len);

        return Collections.max(arr);

    }
}